indexing-service/src/main/java/io/druid/indexing/common/task/IndexTask.java

```diff
@@ -482,7 +487,22 @@ public boolean publishSegments(Set<DataSegment> segments, Object commitMetadata)
         }
       };
 
-      final SegmentsAndMetadata published = driver.finish(publisher, committerSupplier.get());
+      final SegmentsAndMetadata published;
+      final long publishTimeout = ingestionSchema.getTuningConfig().getPublishTimeout();
+      if (publishTimeout == 0) {
+        published = driver.publish(
```

Comment:
I think as per the current KafkaIndexTask implementation the task should just block until publish suceeds